subsys/usb/device_next/usbd_ch9.c

```diff
@@ -913,6 +913,55 @@ static int std_request_to_host(struct usbd_context *const uds_ctx,
 	return ret;
 }
 
+static struct usbd_desc_node *usbd_get_bos_vreq(struct usbd_context *const uds_ctx,
+						const uint8_t vcode)
+{
+	struct usbd_desc_node *desc_nd;
+
+	SYS_DLIST_FOR_EACH_CONTAINER(&uds_ctx->descriptors, desc_nd, node) {
```

Comment:
Why abuse the descriptors list? I think it would be better to just have a separate list for the vendor requests. The vendor requests should not be limited to BOS-only, but generally any vendor (here: application developer) should be able to provide their own vendor commands in application.

Reply:
> Why abuse the descriptors list? 

It is a descriptor and it does not abusing anything.

> I think it would be better to just have a separate list for the vendor requests. The vendor requests should not be limited to BOS-only, but generally any vendor (here: application developer) should be able to provide their own vendor commands in application.

There are no requirements in the specification to allow arbitrary vendor request with recipient device. Using the BOS platform capability is the most portable.

Reply:
How is this a descriptor? This is not descriptor in any way, just a way for application to register vendor commands. USB specification doesn't require the devices to implement any optional functionality (vendor commands are optional functionality). In this context the specification states "A device vendor may also define requests supported by the device.".

What do you mean by "most portable"? What does portability have to do with a stack functionality to allow application developer to support vendor commands?

Reply:
Please familiarize yourself with how MSOS2 and WebUSB use the BOS descriptor, I am pretty sure you will understand it.